tests/foreman/ui/test_activationkey.py

```diff
@@ -269,6 +269,124 @@ def test_positive_create_with_host_collection(self):
                 (strategy, value % host_col.name))
             self.assertIsNotNone(host_collection)
 
+    @tier2
+    def test_positive_add_host_collection_non_admin(self):
+        """Test that hosts can be associated to Activation Keys by non-admin
+        user.
+
+        @id: 417f0b36-fd49-4414-87ab-6f72a09696f2
+
+        @Assert: Activation key is created, added host collection is listed
+
+        @CaseLevel: Integration
+        """
+        name = gen_string(str_type='alpha')
+        # create Host Collection using API
+        host_col = entities.HostCollection(
+            organization=self.organization,
+            name=gen_string(str_type='alpha'),
+        ).create()
+        # Create non-admin user with specified permissions
+        role = entities.Role().create()
+        for res_type in ['Katello::ActivationKey', 'Katello::HostCollection']:
+            permission = entities.Permission(resource_type=res_type).search()
+            entities.Filter(permission=permission, role=role).create()
+        # Add 'Viewer' role to be able to read default environment and
+        # content view while creating AK
+        role2 = entities.Role().search(query={'search': 'name="Viewer"'})[0]
+        password = gen_string('alphanumeric')
+        user = entities.User(
+            admin=False,
+            role=[role, role2],
+            password=password,
+            organization=[self.organization],
+        ).create()
+
+        with Session(
+                self.browser,
+                user=user.login,
+                password=password
+        ) as session:
+            make_activationkey(
```

Comment:
Yeap, tailings after redundant parameters that were not fitting one line were removed. Thanks for pointing out.